joblib/test/test_hashing.py

```diff
@@ -320,3 +322,18 @@ def test_set_hash():
     b = k.f(a)
 
     nose.tools.assert_equal(hash(a), hash(b))
+
+
+def test_string():
+    string = 'foo'
+    a = {string: 'bar'}
+    b = {string: 'bar'}
+    c = pickle.loads(pickle.dumps(b))
+    assert_equal(hash([a, b]), hash([a, c]))
+
+
+def test_dtype():
```

Comment:
Tests that require numpy should be protected with `@with_numpy` to skip them when numpy is not installed:

```
from joblib.test.common import np, with_numpy
```